src/CouchDB.Driver/CouchDatabase.cs

```diff
@@ -500,6 +500,42 @@ public Task DeleteIndexAsync(IndexInfo indexInfo, CancellationToken cancellation
 
         #endregion
 
+        #region View
+
+        /// <inheritdoc/>
+        public Task<CouchViewResult<TValue>> GetViewAsync<TValue>(string design, string view, CouchViewOptions? options = null, CancellationToken cancellationToken = default)
```

Comment:
Should this return just an `IList<TValue>` where TValue is `ICouchDocument`?

Reply:
`TValue` is the value returned from the view which can be any valid object/json in the end. That's why I made the `CouchViewResult` class that contains the result returned by the view. So the user gets the full result and uses what he wants.

Reply:
I say this because the documentation says:
`Array of view row objects. By default, the information returned contains only the document ID and revision.`
But yeah, not needed, also `ICouchDocument` has `Attachments` that is not needed.
However, I still prefer to have a `List<TValue>` because the goal is to help devs as much as possible.

What can be done in this case is to have both:
* `Task<List<TValue>> GetViewAsync(...)` 
* `Task<CouchViewResult<TValue>> GetDetailedViewAsync<TValue>(...)`